indico/modules/events/timetable/util.py

```diff
@@ -196,3 +217,52 @@ def render_session_timetable(session, timetable_layout=None, management=False):
     event_info = serialize_event_info(session.event_new)
     tpl = get_template_module('events/timetable/_timetable.html')
     return tpl.render_timetable(timetable_data, event_info, timetable_layout=timetable_layout, management=management)
+
+
+def get_top_timetable_session_blocks(event, day):
+    """Returns a list of event top-level session blocks for the
+       given `day`
+    """
+    return (event.timetable_entries
+            .filter(db.cast(TimetableEntry.start_dt.astimezone(event.tzinfo), db.Date) == day.date())
+            .filter(TimetableEntry.type == TimetableEntryType.SESSION_BLOCK, TimetableEntry.parent_id.is_(None))
```

Comment:
also, why not pass the second filter as the second argument of the first `filter()` call?